src/Mod/Fem/femsolver/z88/tasks.py

```diff
@@ -117,23 +118,29 @@ def run(self):
         # TODO: search out for "Vector GS" and "Vector KOI" and print values
         # may be compare with the used ones
         self.pushStatus("Executing solver in test mode...\n")
-        self._process = subprocess.Popen(
-            [binary, "-t", "-" + solver],
-            cwd=self.directory,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE)
-        self.signalAbort.add(self._process.terminate)
-        self._process.communicate()
-        self.signalAbort.remove(self._process.terminate)
+        Solve.runZ88("-t", binary, solver, "hide", self)
 
         # run solver real mode
         self.pushStatus("Executing solver in real mode...\n")
-        binary = settings.get_binary("Z88")
-        self._process = subprocess.Popen(
-            [binary, "-c", "-" + solver],
-            cwd=self.directory,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE)
+        # starting normal because the user must see the z88 window
+        Solve.runZ88("-c", binary, solver, "normal", self)
+
+
+    def runZ88(command, binary, solver, state, self):
```

Comment:
```suggestion
    def runZ88(self, command, binary, solver, state):
```
If this hasn't caused you any unexpected results, then the code hasn't been tested properly. (This would always use the wrong object as self.)

Reply:
I am no Pythionist. Why does the order matter here?
I made your change now and it worked as well as before but would like to understand why you want self to be as first argument.

Reply:
I suggest making two changes, actually: as currently written, `runZ88` is being called as a C++ programmer would think of as a _static_ member function, with an additional manually-specified argument specifying an instance to work on (so why is the functions static?!) As originally written, then, the order really doesn't matter because `self` is being passed in completely manually. Calling it like this is _very_ non-idiomatic -- I cannot figure out why it's being done, there seems to be no particular advantage. In my opinion the call itself should be
```
self.runZ88("-c", binary, solver, "normal")
```
which would the necessitate the change suggested by @benj5378, since in this format `self` is passed implicitly to the function as its first argument. Is there some reason to use the `Solve.runZ88("-c", binary, solver, "normal", self)` call format?

Reply:
As I wrote I already adopted  the PR. I understand now that I just need "self" of the class, and as first argument, this is just passed.
I need/use the class' "self" in runZ88() for self._process

Reply:
@donovaly When creating a class, the class itself is always passed as the first parameter which we usually call self. Normally, this happens implicitly as the first argument. So when you write command as the first parameter, then this actually gets passed the class itself while self ends up being the state. `self._process` is expected to be "interpreted" as `Solve._process`, but since command was the first parameter, `command` is in reality the class. And because self was given implicitly, all the parameters shift so with `self.runZ88("-c", binary, solver, "normal")`, `self` would be assigned the string `"normal"`.

Or, this is how it normally goes on. @chennes writes that self is parsed manually so it doesn't have any meaning here - why I don't understand, but I believe he knows what he is talking about.

Reply:
I still don't understand what the problem is. I changed already the argument call order as you requested.

(I googled a bit around and it seems that also my initial PR is valid, only unusal. I also debugged it and the arguments were correctly passed.)

I merged the PR to move on. Feels free to improve the code further.